apps/web/components/connections/ConnectionCard/ConnectionCard.tsx

```diff
@@ -173,18 +174,22 @@ function EditingDisplayName(props: EditingDisplayNameProps) {
       }
     }
 
-    document.addEventListener('keyup', handleKeyUp)
+    document.addEventListener('keydown', handleKeyDown)
     return () => {
-      document.removeEventListener('keyup', handleKeyUp)
+      document.removeEventListener('keydown', handleKeyDown)
     }
   }, [displayName, onCancel, onUpdateSuccess, resourceId, updateResource])
 
   const inputRef = React.useRef<HTMLInputElement>(null)
   useEffect(() => {
-    inputRef.current?.focus()
+    // HACK: don't know why but doing it sync or with 0 timeout doesn't work
+    // using autoFocus props on input also doesn't work. It might have
+    // something to do with radix dropdown focus management.
+    setTimeout(() => inputRef.current?.focus(), 100)
   }, [inputRef])
+
   return (
-    <div className="relative flex grow items-center gap-2 rounded bg-venice-black px-2 py-1">
+    <div className="relative flex grow items-center gap-2 rounded bg-venice-black px-2 py-1 focus-within:ring-1 focus-within:ring-inset focus-within:ring-venice-green">
```

Comment:
Add focus state. Without it, it was confusing whether you have focus on the edit input or not.